src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Hover/HoverFactory.cs

```diff
@@ -224,7 +225,11 @@ internal static class HoverFactory
         ISolutionQueryOperations solutionQueryOperations,
         CancellationToken cancellationToken)
     {
-        var descriptionInfos = descriptors.SelectAsArray(BoundElementDescriptionInfo.From);
+        // Filter out attribute descriptors since we're creating an element hover
+        var keepAttributeInfo = FileKinds.GetFileKindFromFilePath(documentFilePath) == FileKinds.Legacy;
+        var descriptionInfos = descriptors
+            .Where(d => keepAttributeInfo || !d.IsAttributeDescriptor())
```

Comment:
Seems a little wrong to execute the where clause at all when keepAttributeInfo is true, but probably not worse complicating the code since n is going to be low and perf savings small.